### ISSUE TITLE
Don't add unused GridView filters to url

### ISSUE THREAD

**Issue by darkalor** [yiisoft/yii2#10839]
Currently setting any filter in a GridView produces a very messy url that lists all related models and attributes even if there's no actual filter applied there

```
?UserSearch[username]=bob&UserSearch[full_name]=&UserSearch[email]=&UserSearch[gender]=&...
```

Only found a closed/rejected PR https://github.com/yiisoft/yii2/pull/7450 so opening an actual issue, since it would be great to have this improved.

One possible solution to dealing with defaults could be to explicitly track fields there were changed (similar to dirtyAttributes in models) and submit only those. That way filters that were never set are not submitted, and it's still possible to override a default value with an empty filter.

**Comment by schtanislau** [comment]
+1

**Comment by Faryshta** [comment]
I think for this to work we need to specify an url for the grid on which to append the filters.

```php
GridView::widget([
    'dataProvider' => $dataProvider,
    'filterModel' => $searchModel,
    'filterUrl' => ['product/index'],
])
```

The problem will be how to standarize that to work with search using forms, sorting and pagination

**Comment by samdark** [comment]
Won't be implemented in 2.0.
